server/src/test/java/keywhiz/service/permission/AnyPermissionCheckTest.java

```diff
@@ -0,0 +1,74 @@
+package keywhiz.service.permission;
+
+import java.util.Collections;
+import java.util.List;
+import keywhiz.service.permissions.Action;
+import keywhiz.service.permissions.AnyPermissionCheck;
+import keywhiz.service.permissions.PermissionCheck;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class AnyPermissionCheckTest {
+
+  @Rule public MockitoRule mockito = MockitoJUnit.rule();
+  @Mock private PermissionCheck delegate1;
+  @Mock private PermissionCheck delegate2;
+
+  private static Object source;
+  private static Object target;
```

Comment:
I'm torn on whether they should be initialized (if I'm reading correctly, they are not and thus remain null). I can't think of a case where either should legitimately be null, but OTOH maybe the composite should be "dumb" and just strictly delegate to the subordinates (great term, by the way)? I'd be fine leaving as-is.